lesson-4/script.js

```diff
@@ -0,0 +1,74 @@
+const correctResult = (0.1 + 0.2).toFixed(1);
+console.log(correctResult);
+const userLogin = prompt("Enter your login.");
+console.log(userLogin);
+const userPassword = prompt("Enter your password.");
+console.log(userPassword);
+if (userLogin == "admin" && userPassword == "12pass33210") {
+    console.log("Access is allowed.");
+} else if (userLogin != "admin" && userPassword != "12pass33210") {
+    console.log("This user have no access to the panel.");
+} else {
+    console.log("Login or password are wrong.")
+}
+let userMoney = +prompt("How much money do you have?");
+console.log(`User have ${userMoney} UAH`);
+console.log("Here is our price list below:")
+let Watermelon = 30;
+console.log(`Watermelon: ${Watermelon} UAH`);
+let Carrot = 8;
+console.log(`Carrot: ${Carrot} UAH`);
+let Potato = 12;
+console.log(`Potato: ${Potato} UAH`);
+let userChoice = prompt("What would you like to buy?")
+const userChooseWatermelon = userMoney / Watermelon;
+// console.log(Math.floor(userChooseWatermelon));
+const userChooseWatermelonChange = userMoney % Watermelon;
+// console.log(userChooseWatermelonChange);
+const userChooseCarrot = userMoney / Carrot;
+// console.log(Math.floor(userChooseCarrot));
+const userChooseCarrotChange = userMoney % Carrot;
+// console.log(userChooseCarrotChange);
+const userChoosePotato = userMoney / Potato;
+// console.log(Math.floor(userChoosePotato));
+const userChoosePotatoChange = userMoney % Potato;
+// console.log(userChoosePotatoChange);
+if (userChoice == "Watermelon" && userMoney >= Watermelon) {
+    console.log(`За ${userMoney} грн Ви можете придбати ${Math.floor(userChooseWatermelon)} одиниць даного продукту. У Вас залишиться ${userChooseWatermelonChange} грн`);
+} else if (userChoice == "Watermelon" && userMoney < 8) {
```

Comment:
```suggestion
} else if (userChoice === "Watermelon" && userMoney < 8) {
```

Reply:
why we need this type of comparison here? why we need the same data type from user if we waiting just string after prompt?
and in the next examples below

Reply:
Better to use `===` as much as possible. JS is a dynamically typed language. We should remember that